packages/devtools_app/lib/src/memory/memory_tracker_model.dart

```diff
@@ -160,78 +151,142 @@ class TreeTracker {
-    final TrackerAllocation allocation = item.parent;
-
-    // Show the entire call stack for this instance.
-    allocation.displayAll = true;
-
-    // Re-construct the callstack for this instance.
-    final leafNodesCount = allocation.children.length;
-
-    assert(allocation.children.last is TrackerMore);
-    allocation.removeLastChild(); // Remove "More ..."
+  Widget _helpTrackScreen(BuildContext context) {
+    return Column(
+      mainAxisAlignment: MainAxisAlignment.center,
+      children: [
+        const Text('To track allocations for a class, enable the '
+            'checkbox for that class in the table above.'),
+        Row(
+          mainAxisAlignment: MainAxisAlignment.center,
+          children: [
+            const Text('After interacting with your app, come '
+                'back to this tool and click the track button '),
+            trackImage(context),
+          ],
+        ),
+        const Text('to view the collected stack '
+            'traces of constructor calls.'),
+      ],
+    );
+  }
 
-    // Add the entire call stack as leaf nodes.
-    final totalCallStack = item.stacktrace.length;
-    for (var index = leafNodesCount - 1; index < totalCallStack; index++) {
-      allocation.addChild(TrackerCall(item.stacktrace[index]));
-    }
-    allocation.expandCascading();
+  Widget _helpStacktraceScreen(BuildContext context) => Row(
+        mainAxisAlignment: MainAxisAlignment.center,
+        children: const [
+          Text('Select an instance of a tracked class to'
+              ' view its allocation stack trace.'),
+        ],
+      );
+
+  Widget displaySelectedStackTrace(
+    BuildContext context,
+    MemoryController controller,
+    ScrollController scroller,
+    TrackerAllocation tracker,
+  ) {
+    final allocationStacktrace = tracker.stacktrace;
+
+    final stackTrace = allocationStacktrace.stacktrace;
+    final sources = allocationStacktrace.sources;
+    final callstackLength = stackTrace.length;
+    assert(sources.length == callstackLength);
+
+    final titleBackground = titleSolidBackgroundColor(Theme.of(context));
+
+    return Column(
+      children: [
+        SizedBox(
+          width: double.infinity,
+          height: defaultRowHeight + 4, // From table.dart
+          child: Container(
+              color: titleBackground,
+              child: Align(
+                child: Text(
+                  '${tracker.parent.name} Call Stack for Instance ${tracker.index} @ '
+                  '${prettyTimestamp(tracker.timestamp)}',
+                  textAlign: TextAlign.center,
+                  style: TextStyle(backgroundColor: titleBackground),
+                ),
+              )),
+        ),
+        Expanded(
+          child: Scrollbar(
+            isAlwaysShown: true,
+            controller: scroller,
+            child: ListView.builder(
+              controller: scroller,
+              itemCount: callstackLength,
+              itemBuilder: (context, index) {
```

Comment:
there is a named parameter to ListView to add separators between items. probably use it to add a line between each stack frame

Reply:
I used List.separator.